### ISSUE TITLE
Clicking a cell in perlin_noise_1 example causes exception

### ISSUE THREAD

**Issue by DavidAntliff** [pvcraven/arcade#310]
## Bug Report

Clicking a cell in the displayed perlin noise results in an AttributeError exception.

### Actual behavior:

```
Click coordinates: (147, 114). Grid coordinates: (9, 12)
Traceback (most recent call last):
  File "/home/dantliff/.pyenv/versions/3.7.1/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/dantliff/.pyenv/versions/3.7.1/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/arcade/examples/perlin_noise_1.py", line 157, in <module>
    main()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/arcade/examples/perlin_noise_1.py", line 153, in main
    arcade.run()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/arcade/window_commands.py", line 214, in run
    pyglet.app.run()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/app/__init__.py", line 142, in run
    event_loop.run()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/app/base.py", line 175, in run
    self._run()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/app/base.py", line 188, in _run
    platform_event_loop.step(timeout)
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/app/xlib.py", line 125, in step
    device.select()
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/canvas/xlib.py", line 201, in select
    dispatch(e)
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/window/xlib/__init__.py", line 897, in dispatch_platform_event_view
    event_handler(e)
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/window/xlib/__init__.py", line 1202, in _event_button
    self.dispatch_event('on_mouse_press', x, y, button, modifiers)
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/window/__init__.py", line 1313, in dispatch_event
    if EventDispatcher.dispatch_event(self, *args) != False:
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/pyglet/event.py", line 370, in dispatch_event
    if handler(*args):
  File "/home/dantliff/.pyenv/versions/arcade-3.7.1/lib/python3.7/site-packages/arcade/examples/perlin_noise_1.py", line 143, in on_mouse_press
    if self.grid[row][column] == 0:
AttributeError: 'MyGame' object has no attribute 'grid'
```

### Expected behavior:

I'm not sure, but probably not a crash.

### Steps to reproduce/example code:

Run `python -m arcade.examples.perlin_noise_1` and click a cell with the mouse.

**Comment by SirGnip** [comment]
I pushed up a PR with a fix to fix the exception seen when clicking (caused by mixing local and member variable references to "grid").  Clicking with the mouse now doesn't throw an exception, but it also doesn't do anything useful.  The docstring at the top of the script states "TODO: This code doesn't work properly...".  I'm not exactly sure what clicking on a cell is suppose to do.  So, this PR is just an incremental improvement.

**Comment by pvcraven** [comment]
Closed with 2.0 release.
